src/test/groovy/org/osiam/test/integration/RegistrationIT.groovy

```diff
@@ -129,6 +131,17 @@ class RegistrationIT extends AbstractIT {
         Extension extension = user.getExtension('urn:org.osiam:scim:extensions:addon-self-administration')
         extension.getField('activationToken', ExtensionFieldType.STRING) != null
         user.getGroups().get(0).getDisplay().equalsIgnoreCase("Test")
+
+        //Waiting at least 5 seconds to ensure that an email was send
+        Thread.sleep(5000)
```

Comment:
you could actively poll for new emails. No need for a latch, if you don't poll in a background thread.